pkgdb/src/buildenv/realise.cc

```diff
@@ -60,36 +60,171 @@ namespace flox::buildenv {
 #  error "COMMON_NIXPKGS_URL must be set to a locked flakeref of nixpkgs to use"
 #endif
 
-#ifndef FLOX_BASH_BIN
-#  error "FLOX_BASH_BIN must be set to the path of a Bash executable"
+#ifndef FLOX_BASH_PKG
+#  error "FLOX_BASH_PKG must be set to the path of the nix bash package"
+#endif
+
+#ifndef FLOX_COREUTILS_PKG
+#  error \
+    "FLOX_COREUTILS_PKG must be set to the path of the nix coreutils package"
+#endif
+
+#ifndef FLOX_GNUSED_PKG
+#  error "FLOX_GNUSED_PKG must be set to the path of the nix gnused package"
+#endif
+
+#ifndef FLOX_PROCPS_PKG
+#  error "FLOX_PROCPS_PKG must be set to the path of the nix procps package"
 #endif
 
 /* -------------------------------------------------------------------------- */
 
+// Top-level activate script, always invoked with nix bash.
+const char * const ACTIVATE_SCRIPT = R"_(
```

Comment:
I could not agree more, but can we do that in a separate PR? There's a lot of machinery around this that I don't want to touch for fear of triggering a chain of yak-shaving expeditions. I'd also be very happy to work on this.

Reply:
@ysndr I would love to split these out into separate scripts with this PR because I find it incredibly annoying to have to edit these scripts inline, and without the benefit of `shellcheck` for error checking and linting. The concern would be that adding this scope creep to this PR would make it that much harder to merge, but at this point all those script snippets have been rewritten anyway, so I'm starting to think we should. Do you think that we should include a refactoring of these scripts with this PR? If so that will be my next commit ...

Reply:
FTR I looked into this and it quickly grew to the scope of an entirely different PR and code reorganization. What I found was that this activate script, the activate.d directory, and etc-profiles all worked together to act as a "floxEnv library" of sorts, and that if all these were managed as a single package the same environment could be activated using different versions of that package, opening up the possibility of applying updated activation logic to a previously-rendered environment. This is powerful because it means that activation behaviours can improve over time without having to trigger otherwise unnecessary environment rebuilds (not to mention making it much easier to develop/debug/test!).

With that we won't be splitting it out (and redesigning the build accordingly) but I have logged #1343 to remind us to follow up with this and all other issues highlighted by this refactoring effort. Thanks again for the suggestion!